### ISSUE TITLE
OpenMRS - validate mapping on map create

### ISSUE THREAD

**Issue by fatmau** [OpenConceptLab/oclapi#301]
As a User, 
- When I create a Mapping under a Source with OpenMRS validation scheme
- If Mapping contains a duplicate Source-Target Pair
- I should see a validation error "Cannot map concept to itself."

**Comment by djazayeri** [comment]
In an OpenMRS-validated source:

1. I tried to create two mappings between the same concepts, and I am correctly prevented from doing this. The error message is helpful ("There can be only one mapping between two concepts"). (It's a better error message than what is written in the user story comment, so I assume this is correct behavior.)
2. I created a mapping from X to Y, and then created a mapping from Y to X, and this was permitted. (I guess that's also correct.)

In a Basic validation source

* I am correctly allowed to create two mappings with the same source and destination.

(So, I think this passes UAT.)
